### PR TITLE
Support for dtypes different than float32

### PR DESCRIPTION
**Pull request by migperfer**
Some kernel values were forced to float32 previously, now user can use whatever fp they desire. Also fixed the `from scipy import fft` for newer versions of scipy

### CODE REVIEW

File: Installation/nnAudio/utils.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
 import math
 from scipy.signal import get_window
 from scipy import signal
-from scipy import fft
+from scipy.fft import fft
```
Comment (KinWaiCheuk) [inline Installation/nnAudio/utils.py:13]: `from scipy.fftpack import fft` might be even a better choice.

**Review by KinWaiCheuk — CHANGES_REQUESTED** [review]

**Review by KinWaiCheuk — COMMENTED** [review]
Shall we change only line 13? For the fp32 kernels, it is necessary to keep it fp32, otherwise there will be type mismatch error since some kernels created with scipy are fp64.

And I did a quick test on fp16, fp32 support, there is no problem even with the current version of the code. May I know what error did you encounter previously?

**Comment by migperfer** [comment]
Hi! Sorry for the delay, I have been a little bit overwhelmed with work. 

I can't replicate now the error, it's working as you said without changing anything. I could have done something wrong at some other point and debugged down by mistake that "It is the np.float32 fault". Are you still interested into the update to `from scipy.fftpack import fft` anyways?

**Comment by KinWaiCheuk** [comment]
> Hi! Sorry for the delay, I have been a little bit overwhelmed with work.

Don't worry, I was just wondering if you missed my comments or not.

> Are you still interested into the update to `from scipy.fftpack import fft` anyways?
Yes, please :)

File: Installation/nnAudio/utils.py (hunk 1 above)
Comment (migperfer) [reply Installation/nnAudio/utils.py]: Done!

**Comment by migperfer** [comment]
Done! Think should be working now. Sorry for the long time I've been away.

**Review by KinWaiCheuk — APPROVED** [review]

**Pull request merged**